commit 72cbad117d0abb4b47a5d00754ea465eda6c06b5
Author: Lamarque V. Souza <lamarque@gmail.com>
Date:   Sun Apr 10 18:34:17 2011 -0300

    Fix Solid::Networking::status() returning Solid::Networking::Unknown
    sometimes.
    
    BUG: 270538
    FIXED-IN: 4.6.3

diff --git a/solid/solid/networking.cpp b/solid/solid/networking.cpp
index 6dd6553..02c7714 100644
--- a/solid/solid/networking.cpp
+++ b/solid/solid/networking.cpp
@@ -59,6 +59,7 @@ Solid::Networking::Notifier::Notifier()
 void Solid::NetworkingPrivate::initialize()
 {
     QDBusPendingReply<uint> reply = iface->status();
+    reply.waitForFinished();
     if (reply.isValid()) {
         netStatus = ( Solid::Networking::Status )reply.value();
     } else {
